data-raw/qs_metabolic.R

```diff
@@ -0,0 +1,105 @@
+#' or companies wanting to use the CoEQ should request a license to do so from the following
+#' [link](https://licensing.leeds.ac.uk/product/control-of-eating-questionnaire-coeq).
+
+# Load libraries ----
+
+library(tibble)
+library(dplyr)
+library(pharmaversesdtm)
+library(purrr)
+library(stringr)
+
+# Set seed for random data generation ----
+set.seed(3.14159265)
+
+# Read input data ----
+
+vs_metabolic <- admiralmetabolic::vs_metabolic
+
+# Set up questions, codes and categories ----
+coeq_structure <- tibble::tribble(
+  ~QSCAT, ~QSSCAT, ~QSTESTCD, ~QSTEST,
+  "COEQ", "EATING", "COEQ01", "How hungry have you felt?",
+  "COEQ", "EATING", "COEQ02", "How full have you felt?",
+  "COEQ", "EATING", "COEQ03", "How strong was your desire to eat sweet foods?",
+  "COEQ", "EATING", "COEQ04", "How strong was your desire to eat savoury foods?",
+  "COEQ", "EMOTIONS", "COEQ05", "How happy have you felt?",
+  "COEQ", "EMOTIONS", "COEQ06", "How anxious have you felt?",
+  "COEQ", "EMOTIONS", "COEQ07", "How alert have you felt?",
+  "COEQ", "EMOTIONS", "COEQ08", "How contented have you felt?",
+  "COEQ", "GENERAL CRAVINGS", "COEQ09", "During the last 7 days how often have you had food cravings?",
+  "COEQ", "GENERAL CRAVINGS", "COEQ10", "How strong have any food cravings been?",
+  "COEQ", "GENERAL CRAVINGS", "COEQ11", "How difficult has it been to resist any food cravings?",
+  "COEQ", "GENERAL CRAVINGS", "COEQ12", "How often have you eaten in response to food cravings?",
+  "COEQ", "GENERAL CRAVINGS", "COEQ13", "Chocolate or chocolate flavoured foods",
+  "COEQ", "SPECIFIC CRAVINGS", "COEQ14", "Other sweet foods (cakes, pastries, biscuits, etc)",
+  "COEQ", "SPECIFIC CRAVINGS", "COEQ15", "Fruit or fruit juice",
+  "COEQ", "SPECIFIC CRAVINGS", "COEQ16", "Dairy foods (cheese, yoghurts, milk, etc)",
+  "COEQ", "SPECIFIC CRAVINGS", "COEQ17", "Starchy foods (bread, rice, pasta, etc)",
+  "COEQ", "SPECIFIC CRAVINGS", "COEQ18", "Savoury foods (french fries, crisps, burgers, pizza, etc)",
+  "COEQ", "CONTROL EATING", "COEQ19", "Generally, how difficult has it been to control your eating?",
+  "COEQ", "CONTROL EATING", "COEQ20", "Which one food makes it most difficult for you to control eating?",
+  "COEQ", "CONTROL EATING", "COEQ21", "How difficult has it been to resist eating this food during the last 7 days?",
+)
+
+# Use visit schedule and days from VS ----
+visit_schedule <- vs_metabolic %>%
+  select(STUDYID, USUBJID, VISIT, VISITNUM, VISITDY, VSDTC, VSDY) %>%
+  filter(str_detect(VISIT, "AMBUL|RETRIEVAL", negate = TRUE)) %>%
+  rename(QSDTC = VSDTC, QSDY = VSDY) %>%
+  distinct()
+
+# Cross join to get questions at each visit ----
+qs_metabolic_shell <- visit_schedule %>%
+  cross_join(coeq_structure)
+
+# Simulate question answers ----
+qs_metabolic_results <- qs_metabolic_shell %>%
+  mutate(
+    DOMAIN = "QS",
+    QSORRES = floor(runif(1) * 10) + round(runif(1)),
```

Comment:
Values should be from 0 to 100, apart from question 20 which is a free text question.

Reply:
When I run this code it is not a random number that is generated but all rows get a value of QSORRES=2.

Reply:
```suggestion
    QSORRES = floor(runif(n()) * 10) + round(runif(n())),
```

Reply:
@kathrinflunkert shouldn't we have a 0-10 scale for individual questions, and in the ADQS a 0-100 scale for subscales?

Reply:
@starosto I am still trying to gather some more information about the different versions of the CoEQ, but the 21-item version for which we got the license has a 0 to 100 mm VAS scale:

  <img width="455" alt="image" src="https://github.com/user-attachments/assets/57b24b94-702d-4a51-b019-2132b343e99e">



There is apparently another version where the user is only able to tick values between 0 and 10, but for this version I cannot find where/how to license it:
 
<img width="476" alt="image" src="https://github.com/user-attachments/assets/cc2efd45-8bdb-45a2-bf29-68bdf1cb8f56">

Reply:
@kathrinflunkert I see, I didn't know about the CoEQ VAS scale. Thanks for the clarification! :)